src/pyvesync/vesyncfan.py

```diff
@@ -2010,13 +2017,437 @@ def set_display(self, mode: bool) -> bool:
+
+        body['payload']['data'] = {
+            'screenSwitch': int(mode)
+        }
+
+        r, _ = Helpers.call_api(
+            '/cloud/v2/deviceManaged/bypassV2',
+            method='post',
+            headers=head,
+            json_object=body,
+        )
+
+        if r is not None and Helpers.code_check(r):
+            return True
+        logger.debug("Error toggling display - %s", self.device_name)
+        return False
+
+    def turn_on_display(self) -> bool:
+        """Turn display on."""
+        return self.set_display_enabled(True)
+
+    def turn_off_display(self):
+        """Turn display off."""
+        return self.set_display_enabled(False)
+
+    def set_humidity(self, humidity: int) -> bool:
+        """Set target humidity for humidity mode."""
+        if humidity < 30 or humidity > 80:
+            logger.debug("Humidity value must be set between 30 and 80")
+            return False
+        head, body = self.build_api_dict('setTargetHumidity')
+
+        if not head and not body:
+            return False
+
+        body['payload']['data'] = {
+            'targetHumidity': humidity
+        }
+
+        r, _ = Helpers.call_api(
+            '/cloud/v2/deviceManaged/bypassV2',
+            method='post',
+            headers=head,
+            json_object=body,
+        )
+
+        if r is not None and Helpers.code_check(r):
+            return True
+        logger.debug('Error setting humidity')
+        return False
+
+    def set_humidity_mode(self, mode: str) -> bool:
+        """Set humidifier mode."""
+        if mode not in self.mist_modes:
+            logger.debug('Invalid humidity mode used - %s',
+                         mode)
+            logger.debug('Proper modes for this device are - %s',
+                         str(self.mist_modes))
+            return False
+        head, body = self.build_api_dict('setHumidityMode')
+        if not head and not body:
+            return False
+        body['payload']['data'] = {
+            'workMode': 'autoPro' if mode == 'auto' else mode
+        }
+
+        r, _ = Helpers.call_api(
+            '/cloud/v2/deviceManaged/bypassV2',
+            method='post',
+            headers=head,
+            json_object=body,
+        )
+
+        if r is not None and Helpers.code_check(r):
+            return True
+        logger.debug('Error setting humidity mode')
+        return False
+
+    def set_auto_mode(self) -> bool:
+        """Set humidity mode to auto."""
+        return self.set_humidity_mode('auto')
+
+    def set_manual_mode(self) -> bool:
+        """Set humidity mode to manual."""
+        return self.set_humidity_mode('manual')
+
+    def automatic_stop_on(self) -> bool:
+        """Set humidity mode to auto."""
+        return self.set_humidity_mode('auto')
+
+    def automatic_stop_off(self) -> bool:
+        """Set humidity mode to manual."""
+        return self.set_humidity_mode('manual')
```

Comment:
I don't have much context on these since I only have the one humidifier. I believe the only way to set an automatic stop is by using the 'humidity' mode, and the humidifier will turn off at the target humidity. I don't see any other method for this behavior.

Reply:
if so, this should be `set_humidity_mode('humidity')`. I will update if that's the case.